setuptools/wheel.py

```diff
@@ -124,3 +131,14 @@ def raw_req(req):
                 os.rmdir(subdir)
             if os.path.exists(dist_data):
                 os.rmdir(dist_data)
+            # Fix namespace packages.
+            namespace_packages = os.path.join(egg_info, 'namespace_packages.txt')
```

Comment:
Apropos @reinout's [comment](https://github.com/pypa/setuptools/issues/1214#issuecomment-348449705) on #1214, shouldn't the line above be like this instead?

    try:
        namespace_packages = zf.read('%s/namespace_packages.txt).splitlines()
    except KeyError:
        pass
    else:
        for mod in namespace_packages:
            [...]

Reply:
Your code isn't completely clear. The `%s`, what is that?

The problem is that `egg_info` instead of `*dist-info/` is used (I believe).

Reply:
The dist-info directory is renamed to EGG-INFO as part of the conversion to an unzipped-egg.

Reply:
Normally, buildout lets setuptools download the release and then calls `setuptools.archive_util.unpack_archive()` when it gets an `.egg` file from setuptools download step.

Now suddenly setuptools starts feeding buildout wheels :-) I'm calling `unpack_archive()` on those, too. It sounds like I should use a different function?

Reply:
> Your code isn't completely clear. The %s, what is that?

@reinout, good catch. I've fixed my comment. that part should be:

    '%s/namespace_packages.txt' % (dist_info,)

The gist of it is that we should read `namespace_packages.txt` from the original zipped wheel metadata, instead of the converted metadata.

Reply:
@leorochael: yes, that ought to help get the `__init__.py` generated, even when I'm using the regular `.unpack_archive()` function.
(I don't know if there's more that we're missing if we don't let setuptools convert it to an egg).

Reply:
Why? This particular metadata is unchanged during the conversion.

Reply:
Like I just said in #1214, I was simply calling the wrong function. I'm letting setuptools extract the .whl now and it does the right thing.